private/buf/bufworkspace/workspace_dep_manager_provider.go

```diff
@@ -82,12 +81,14 @@ func (w *workspaceDepManagerProvider) GetWorkspaceDepManager(
 	bucketTargeting buftarget.BucketTargeting,
 ) (_ WorkspaceDepManager, retErr error) {
 	controllingWorkspace := bucketTargeting.ControllingWorkspace()
-	if controllingWorkspace == nil || controllingWorkspace.BufWorkYAMLFile() != nil {
-		return nil, syserror.New("no supported workspace found")
+	if controllingWorkspace != nil && controllingWorkspace.BufYAMLFile() != nil {
+		// A v2 workspace was found, but we make sure
+		bufYAMLFile := controllingWorkspace.BufYAMLFile()
+		if bufYAMLFile.FileVersion() == bufconfig.FileVersionV2 {
+			return newWorkspaceDepManager(bucket, controllingWorkspace.Path(), true), nil
+		}
 	}
-	bufYAMLFile := controllingWorkspace.BufYAMLFile()
-	if bufYAMLFile.FileVersion() == bufconfig.FileVersionV2 {
-		return newWorkspaceDepManager(bucket, controllingWorkspace.Path(), true), nil
-	}
-	return newWorkspaceDepManager(bucket, bucketTargeting.InputDirPath(), false), nil
+	// Otherwise we simply ignore any buf.work.yaml that was found and attempt to build
+	// a v1 module at the SubDirPath
+	return newWorkspaceDepManager(bucket, bucketTargeting.SubDirPath(), false), nil
```

Comment:
This is because `GetWorkspaceDepManager` only works on v2 workspaces or v1 modules (i.e., not v1 workspaces)?

Reply:
Correct -- yeah, so we gate certain actions (`mod/dep update`, `push`, etc.) from being performed on `v1` workspaces, but `v1` modules are valid. This is done through an option.

https://github.com/bufbuild/buf/blob/c9ad23f6bc66b22f0d5683bea5c92e69e097c7a4/private/buf/bufworkspace/workspace_dep_manager_provider.go#L34-L41